src/iexamine.cpp

```diff
@@ -116,164 +111,135 @@ void iexamine::atm(player *p, map *m, int examx, int examy)
         }
 
-        pos2 = g->inv(_("Insert card for withdrawal."));
-        with = &(p->i_at(pos2));
-        if (with->is_null()) {
-            popup(_("You do not have that item!"));
+        auto const src = choose_card(_("Insert card for withdrawal."));
+        if (!src) {
             return;
-        }
-        if (with->type->id != "cash_card") {
-            popup(_("Please insert cash cards only!"));
-            return;
-        }
-        if (with == dep) {
+        } else if (dst == src) {
             popup(_("You must select a different card to move from!"));
             return;
-        }
-        if (with->charges == 0) {
+        } else if (!src->charges) {
             popup(_("You can only move money from charged cash cards!"));
             return;
         }
 
-        max = with->charges;
-        std::string popupmsg = string_format(ngettext("Transfer how much? Max:%d cent. (0 to cancel) ",
-                                             "Transfer how much? Max:%d cents. (0 to cancel) ",
-                                             max),
-                                             max);
-        amount = std::atoi( string_input_popup( popupmsg, 20,
-                                 to_string(max), "", "", -1, true).c_str()
-                               );
-        if (amount <= 0) {
+        auto const max = src->charges;
+        auto const amount = prompt_for_amount(ngettext(
+            "Transfer how much? Max: %d cent. (0 to cancel) ",
+            "Transfer how much? Max: %d cents. (0 to cancel) ", max), max);
+
+        if (!amount) {
             return;
         }
-        if (amount > max) {
-            amount = max;
-        }
-        with->charges -= amount;
-        dep->charges += amount;
-        p->moves -= 100;
-        return;
 
+        src->charges -= amount;
+        dst->charges += amount;
+        p->moves     -= move_cost;
     } else if (choice == purchase_cash_card) {
         if(query_yn(_("This will automatically deduct $1.00 from your bank account. Continue?"))) {
             item card("cash_card", calendar::turn);
             card.charges = 0;
             p->i_add(card);
-            p->cash -= 100;
-            p->moves -= 100;
+            p->cash  -= card_cost;
+            p->moves -= move_cost;
         }
     } else if (choice == transfer_all_money) {
-        pos = g->inv(_("Insert card for bulk deposit."));
-        dep = &(p->i_at(pos));
-        if (dep->is_null()) {
-            popup(_("You do not have that item!"));
-            return;
-        }
-        if (dep->type->id != "cash_card") {
-            popup(_("Please insert cash cards only!"));
+        auto const dst = choose_card(_("Insert card for bulk deposit."));
+        if (!dst) {
             return;
         }
 
-        //for all cash cards in inventory
+        // sum all cash cards in inventory
+        auto sum = dst->charges;
         for (auto &elem : g->u.inv.all_items_by_type("cash_card")) {
```

Comment:
I know this isn't your code, but doesn't `p->charges_of("cash_card")` (used earlier) return the sum of cash card charges in the player inventory already? It seemed to be used that way.